presto-spi/src/main/java/com/facebook/presto/spi/function/BuiltInFunction.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.function;
+
+public abstract class BuiltInFunction
+        implements SqlFunction
+{
+    @Override
+    public boolean isCalledOnNullInput()
+    {
+        return false;
+    }
+
+    @Override
+    public BoundFunction bind(Signature boundSignature)
```

Comment:
Who calls this? This API looks unnecessary. Even it is, at the very least this needs to check that the given boundSignature actually does not have variable constraints and is compatible with the original signature.

Reply:
Removed